network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/configuration/WgVpnControllerService.kt

```diff
@@ -18,12 +18,85 @@ package com.duckduckgo.networkprotection.impl.configuration
 
 import com.duckduckgo.anvil.annotations.ContributesServiceApi
 import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.di.scopes.VpnScope
+import com.duckduckgo.mobile.android.vpn.service.VpnSocketProtector
+import com.duckduckgo.networkprotection.impl.di.ProtectedVpnControllerService
+import com.duckduckgo.networkprotection.impl.di.UnprotectedVpnControllerService
+import com.squareup.anvil.annotations.ContributesTo
+import dagger.Lazy
+import dagger.Module
+import dagger.Provides
+import dagger.SingleInstanceIn
+import java.security.KeyStore
+import java.security.Security
+import javax.inject.Named
+import javax.net.ssl.SSLSocket
+import javax.net.ssl.TrustManagerFactory
+import javax.net.ssl.X509TrustManager
+import logcat.logcat
+import okhttp3.OkHttpClient
+import org.conscrypt.Conscrypt
+import retrofit2.Retrofit
 import retrofit2.http.Body
 import retrofit2.http.GET
 import retrofit2.http.Headers
 import retrofit2.http.POST
 
+@Module
+@ContributesTo(scope = VpnScope::class)
+object WgVpnControllerServiceModule {
+
+    @Provides
+    @SingleInstanceIn(VpnScope::class)
+    fun provideDelegatingSSLSocketFactory(
+        socketProtector: Lazy<VpnSocketProtector>,
+        @Named("api") okHttpClient: Lazy<OkHttpClient>,
+    ): DelegatingSSLSocketFactory {
+        return object : DelegatingSSLSocketFactory(okHttpClient.get().sslSocketFactory) {
+            override fun configureSocket(sslSocket: SSLSocket): SSLSocket {
+                logcat { "aitor protecting sslSocket" }
+                socketProtector.get().protect(sslSocket)
+                return sslSocket
+            }
+        }
+    }
+
+    @Provides
+    @UnprotectedVpnControllerService
+    @SingleInstanceIn(VpnScope::class)
+    fun providesWgVpnControllerService(
```

Comment:
The `UnprotectedVpnControllerService` is a retrofit API for the controller API that will never to through the VPN. 

In fact, we could totally remove the `ProtectedVpnControllerService` as currently there is no use case for calling the controller API through the VPN tun interface